packages/atomic/src/components/common/validate-props-controller/validate-props-controller.ts

```diff
@@ -0,0 +1,58 @@
+import type {Schema} from '@coveo/bueno';
+import type {ReactiveController, ReactiveControllerHost} from 'lit';
+import type {LitElementWithError} from '@/src/decorators/types';
+
+export class ValidatePropsController implements ReactiveController {
```

Comment:
```suggestion
export class ValidatePropsController<TProps extends Record<string, unknown>>
  implements ReactiveController
{
```

Reply:
Instead of using the `Object` type, let's use the type from the schema and have some nice auto-completion
<img width="392" height="171" alt="image" src="https://github.com/user-attachments/assets/f9ee3719-5f83-4065-b320-d3df265b0b12" />